source/sigv4.c

```diff
@@ -2741,27 +2787,36 @@ static SigV4Status_t generateSigningKey( const SigV4Parameters_t * pSigV4Params,
     SigV4Status_t returnStatus = SigV4Success;
     int32_t hmacStatus = 0;
     char * pSigningKeyStart = NULL;
+    bool isBufferSpaceSufficient = true;
```

Comment:
Is an extra flag really necessary here? One can just check the current value of `returnStatus`.